### PR TITLE
*: add observations client, simple pages to view them

### PR DESCRIPTION
**Pull request by stevekuznetsov**
CONTRIBUTING.md: record what's needed for a dev env

Without the Sentry data, a bunch of errors pop up on every app open.

Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

---

App.tsx: use shared state for center, date

This way, choosing a new center in the dev picker influences where
observations, weather data, etc are loaded from.

Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

---

types/nac: update a field to nullable

We don't use this yet, but apparently it's nullable.

Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

---

*: add observations client, simple pages to view them

This commit adds in:

 - a GraphQL schema dumped from introspecting the endpoints the NAC
   provides
 - auto-generated TypeScript types for the above scehma
 - semi-generated axios- and react-query-based clients
 - a simple scroll-view for observations
 - a page to inspect a specific observation

Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

---

cc @floatplane 

The views are very rudimentary to the point of almost being useless, was mostly just focused on the plumbing here.

### CODE REVIEW

File: App.tsx [hunk 1]
```diff
@@ -219,29 +199,23 @@ const App = () => {
           <SafeAreaProvider>
             <NavigationContainer>
               <SafeAreaView style={styles.container}>
-                <Tab.Navigator initialRouteName="Home" screenOptions={{headerShown: false}}>
```
Comment (stevekuznetsov) [inline App.tsx:222]: I'm now realizing if we had the center picker navigate to a view that has this whole thing as a child, then we are going to have a consistent route data passing paradigm .. ?

Comment (stevekuznetsov) [reply App.tsx:222]: Hm no I can't imagine adding new tabs to one global navigator with the same name but different parameters would be a good idea.

**Comment by stevekuznetsov** [comment]
Looks like Brian's out until January, I will just merge this in and we can iterate after the new year.

File: hooks/observations-fetcher.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+
+import axios from 'axios';
+
+import {ClientContext, ClientProps} from '../clientContext';
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+export const useFetch = <TData, TVariables>(query: string, options?: RequestInit['headers']): ((variables?: TVariables) => Promise<TData>) => {
```
Comment (floatplane) [inline hooks/observations-fetcher.ts:8]: @stevekuznetsov can we rename this file to `useFetch.ts`?

**Pull request merged**